packages/core/src/fhir-to-cda/cda-templates/components/allergies.ts

```diff
@@ -0,0 +1,271 @@
+    },
+  };
+}
+
+function createEntryRelationship(
+  allergy: AllergyIntolerance,
+  referenceId: string
+): ObservationEntryRelationship {
+  const codeSystem = allergy.code?.coding?.[0]?.system;
+  const systemIsLoinc = isLoinc(codeSystem);
+  return {
+    _typeCode: "SUBJ",
+    observation: {
+      _classCode: "OBS",
+      _moodCode: "EVN",
+      templateId: buildInstanceIdentifier({
+        root: oids.allergyIntoleranceObservation,
+        extension: extensionValue2014,
+      }),
+      id: buildInstanceIdentifier({
+        root: placeholderOrgOid,
+        extension: allergy.id,
+      }),
+      code: buildCodeCe({
+        code: allergy.code?.coding?.[0]?.code,
+        codeSystem: systemIsLoinc ? loincCodeSystem : codeSystem,
+        codeSystemName: systemIsLoinc ? loincSystemName : undefined,
+        displayName: allergy.code?.coding?.[0]?.display,
+      }),
+      value: buildValueCd(allergy.code, referenceId),
+      participant: createParticipant(allergy.reaction?.[0], referenceId),
+      entryRelationship: [createReactionEntryRelationship(allergy, referenceId)],
+    },
+  };
+}
+
+function createParticipant(
+  reaction: AllergyIntoleranceReaction | undefined,
+  referenceId: string
+): Participant | undefined {
+  if (!reaction) return undefined;
+  return {
+    _typeCode: "CSM",
+    _contextControlCode: "OP",
+    participantRole: {
+      _classCode: "MANU",
+      playingEntity: {
+        _classCode: "MMAT",
+        code: buildCodeCvFromCodeableConcept(reaction.substance, `${referenceId}-substance`),
+      },
+    },
+  };
+}
+
+function createReactionEntryRelationship(
+  allergy: AllergyIntolerance,
+  referenceId: string
+): ObservationEntryRelationship {
+  const codeSystem = allergy.code?.coding?.[0]?.system;
+  const systemIsLoinc = isLoinc(codeSystem);
+  return {
+    _typeCode: "MFST",
+    _inversionInd: true,
+    observation: {
+      _classCode: "OBS",
+      _moodCode: "EVN",
+      templateId: buildInstanceIdentifier({
+        root: oids.reactionObservation,
+        extension: extensionValue2014,
+      }),
+      id: { _nullFlavor: "NI" },
+      code: buildCodeCe({
+        code: allergy.reaction?.[0]?.manifestation?.[0]?.coding?.[0]?.code,
+        codeSystem: systemIsLoinc ? loincCodeSystem : codeSystem,
+        codeSystemName: systemIsLoinc ? loincSystemName : undefined,
+        displayName: allergy.reaction?.[0]?.manifestation?.[0]?.coding?.[0]?.display,
+      }),
+      text: {
+        reference: {
+          _value: `${referenceId}-reaction`,
+        },
+      },
+      value: buildValueCd(allergy.code, `${referenceId}-reaction`),
+    },
+  };
+}
+
+/**
+ * For FHIR statuses:
+ * @see https://hl7.org/fhir/R4/valueset-allergyintolerance-clinical.html
+ * For CDA statuses:
+ * @see https://terminology.hl7.org/5.2.0/ValueSet-v3-ActStatus.html
+ */
+function mapAllergyStatusCode(coding: Coding[] | undefined): string | undefined {
```

Comment:
Are we not building a type for Allergy status code? We're just returning `string`

Reply:
Ya, these mapping functions are only transforming the string representation of a FHIR status to the CDA status. And most statusCodes are of type `string` anyway, i.e. 
<statusCode code="something">